build.gradle

```diff
@@ -1,3 +1,320 @@
+        }
+        compile "net.sf.jopt-simple:jopt-simple:$joptVersion"
+        compile 'org.fxmisc.easybind:easybind:1.0.3'
+        compileOnly "org.projectlombok:lombok:$lombokVersion"
+        annotationProcessor "org.projectlombok:lombok:$lombokVersion"
+        testCompile 'org.jmockit:jmockit:1.30' // must not use current $jmockitVersion
+        testCompileOnly "org.projectlombok:lombok:$lombokVersion"
+        testAnnotationProcessor "org.projectlombok:lombok:$lombokVersion"
+    }
+}
+
+
+configure(project(':core')) {
+    dependencies {
+        compile project(':assets')
+        compile project(':p2p')
+        compile "net.sf.jopt-simple:jopt-simple:$joptVersion"
+        compile('network.bisq.btcd-cli4j:btcd-cli4j-core:3864e1c4') {
+            exclude(module: 'slf4j-api')
+            exclude(module: 'httpclient')
+            exclude(module: 'commons-lang3')
+            exclude(module: 'jackson-core')
+            exclude(module: 'jackson-annotations')
+            exclude(module: 'jackson-databind')
+        }
+        compile('network.bisq.btcd-cli4j:btcd-cli4j-daemon:3864e1c4') {
+            exclude(module: 'slf4j-api')
+            exclude(module: 'httpclient')
+            exclude(module: 'commons-lang3')
+            exclude(module: 'jackson-core')
+            exclude(module: 'jackson-annotations')
+            exclude(module: 'jackson-databind')
+        }
+        compile 'com.fasterxml.jackson.core:jackson-core:2.8.10'
+        compile 'com.fasterxml.jackson.core:jackson-annotations:2.8.10'
+        compile('com.fasterxml.jackson.core:jackson-databind:2.8.10') {
+            exclude(module: 'jackson-annotations')
+        }
+
+        compileOnly "org.projectlombok:lombok:$lombokVersion"
+        annotationProcessor "org.projectlombok:lombok:$lombokVersion"
+
+        testCompile "org.jmockit:jmockit:$jmockitVersion"
+        testCompile("org.mockito:mockito-core:$mockitoVersion") {
+            exclude(module: 'objenesis')
+        }
+        testCompile "org.powermock:powermock-module-junit4:$powermockVersion"
+        testCompile "org.powermock:powermock-api-mockito2:$powermockVersion"
+        testCompile "org.springframework:spring-test:$springVersion"
+        testCompile "com.natpryce:make-it-easy:$easyVersion"
+        testCompile group: 'org.hamcrest', name: 'hamcrest-all', version: '1.3'
+        testCompileOnly "org.projectlombok:lombok:$lombokVersion"
+        testAnnotationProcessor "org.projectlombok:lombok:$lombokVersion"
+    }
+
+    test {
+        systemProperty 'jdk.attach.allowAttachSelf', true
+
+        def jmockit = configurations.testCompile.files.find { it.name.contains("jmockit") }.absolutePath
+        jvmArgs "-javaagent:$jmockit"
+    }
+}
+
+
+configure(project(':desktop')) {
+    apply plugin: 'com.github.johnrengelman.shadow'
+    apply plugin: 'witness'
+    apply from: '../gradle/witness/gradle-witness.gradle'
+
+    version = '0.8.0-SNAPSHOT'
```

Comment:
A comment that is out of scope for this PR. But perhaps this should be taken out of the gradle file and put into a resource file, similar to how it is done for pricenode?
`version = file("src/main/resources/version.txt").text`

Reply:
Yeah, I cleaned up a number of unnecessary, unused, or just wrong instances of version metadata elsewhere in this PR, but in this case, the assignment of the `$project.version` variable affects the name of the resulting jar and shadow jar, so it's not quite as simple as extracting it to a resource file. There is also a lot of duplication of the current version throughout the codebase and supporting shell scripts that are used during releases. As you mention, it's a topic for another, more comprehensive PR that really cleans this stuff up.